File: project64-scripts/Watches.js
var xAddr = 0x800F69A4
var xVelocityAddr = 0x800F69C4
var yAddr = 0x800F69A8
var yVelocityAddr = 0x800F69C8
var zAddr = 0x800F69AC
var zVelocityAddr = 0x800F69CC
var angleAddr = 0x800F69BC
var angleVelocityAddr = 0x800F6A24
var u1Addr = 0x800F6A28
var accCntAddr = 0x800F6A2C
var powerCheckAddr = 0x8015AABC

var x
var xVelocity
var y
var yVelocity
var z
var zVelocity
var angle
var angleVelocity
var acceleration
var powerCheck

var textHeight = 13
var topTextY = 80

events.ondraw(function () {
    x = mem.float[xAddr]
    xVelocity = mem.float[xVelocityAddr]
    y = mem.float[yAddr]
    yVelocity = mem.float[yVelocityAddr]
    z = mem.float[zAddr]
    zVelocity = mem.float[zVelocityAddr]
    angle = mem.u32[angleAddr] * 180 * Math.pow(2, -15)
    angleVelocity = mem.float[angleVelocityAddr]
    acceleration = mem.float[accCntAddr]
    powerCheck = Math.pow(acceleration, 2) / 25

    var i = 0
    screen.print(0, calculateTextPosition(i++), 'X:              ' + x)
    screen.print(0, calculateTextPosition(i++), 'Y:              ' + y)
    screen.print(0, calculateTextPosition(i++), 'Z:              ' + z)
    screen.print(0, calculateTextPosition(i++), 'X Velocity:     ' + xVelocity)
    screen.print(0, calculateTextPosition(i++), 'Y Velocity:     ' + yVelocity)
    screen.print(0, calculateTextPosition(i++), 'Z Velocity:     ' + zVelocity)
    screen.print(0, calculateTextPosition(i++), 'Angle:          ' + angle)
    screen.print(0, calculateTextPosition(i++), 'Angle Velocity: ' + angleVelocity)
    screen.print(0, calculateTextPosition(i++), 'Acceleration:   ' + acceleration)
    screen.print(0, calculateTextPosition(i++), 'Power Check:    ' + powerCheck)
})

function calculateTextPosition(i) {
    return topTextY + textHeight * i
}

File: bizhawk-scripts/Watches.lua
local xAddr = 0xF69A4
local xVelocityAddr = 0xF69C4
local yAddr = 0xF69A8
local yVelocityAddr = 0xF69C8
local zAddr = 0xF69AC
local zVelocityAddr = 0xF69CC
local angleAddr = 0xF69BC
local angleVelocityAddr = 0xF6A24
local accCntAddr = 0xF6A2C

local x
local xVelocity
local y
local yVelocity
local z
local zVelocity
local angle
local angleVelocity
local accel

while true do
    x = mainmemory.readfloat(xAddr, true)
    xVelocity = mainmemory.readfloat(xVelocityAddr, true)
    y = mainmemory.readfloat(yAddr, true)
    yVelocity = mainmemory.readfloat(yVelocityAddr, true)
    z = mainmemory.readfloat(zAddr, true)
    zVelocity = mainmemory.readfloat(zVelocityAddr, true)
    angle = mainmemory.read_u32_be(angleAddr)
    angleVelocity = mainmemory.readfloat(angleVelocityAddr, true)
    accel = mainmemory.readfloat(accCntAddr, true)
    gui.text(0, 100, "X: " .. x)
    gui.text(0, 120, "Y: " .. y)
    gui.text(0, 140, "Z: " .. z)
    gui.text(0, 160, "xVelocity: " .. xVelocity)
    gui.text(0, 180, "yVelocity: " .. yVelocity)
    gui.text(0, 200, "zVelocity: " .. zVelocity)
    gui.text(0, 220, "Angle: " .. angle * 180 * 2 ^ -15)
    gui.text(0, 240, "AngleVelocity: " .. angleVelocity)
    gui.text(0, 260, "Acceleration: " .. accel)
    gui.text(0, 280, "PowerCheck: " .. accel ^ 2 / 25)
    emu.frameadvance()
end